crates/commitlog/src/commitlog.rs

```diff
@@ -496,8 +504,11 @@ impl<R: Repo> Commits<R> {
         // interesting.
         let prev_error = self.last_error.take();
 
+        // Skip entries before the initial commit.
+        if self.last_commit.offset_before_initial(&commit.min_tx_offset) {
```

Comment:
Ah! I see problem with skipping commits, you will be able to jump to the tx falling in middle of some commit.

Reply:
but that will not work with current implementation as well. Will report `out-of-order`.
So I am confused whether the change to make `CommitlogInfo::Initial` any `commit.min_tx_offset` rather `segment.min_tx_offset` is Correct? @kim 
Looks fine to me.

Reply:
Went little dirty to detect and adjust offset to commit boundary inside the iterator itself.